### PR TITLE
Tickets/dm 30343

### PR DESCRIPTION
**Pull request by pkubanek**
Document Thermal FPGA.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,27 +1,266 @@
...
+(writing) code works by reading (without timeout) from FIFO, and if something
+is available, act accordingly. This is coupled with DIO states (e.g. to make
+sure next bit/byte on serial line is transmitted after the current).
+
+## Command multiplexing
+
+Commands, followed by arguments are filled into _Software
+Resources/CommandFIFO_. _Commands/CommandFIFOMultiplexer_ read this and
+multiplex the command to various handlers _(This is exactly how CPUs are
+handling instructions from binary code)_. Handlers fills in queues (e.g. ModBus
```
Comment (couger01) [inline README.md]:
```suggestion
handle instructions from binary code)_. Handlers fills in queues (e.g. ModBus

```

File: README.md [hunk 2]
```diff
@@ -1,27 +1,266 @@
...
+Receiving (reading) code works by placing values into FIFO. Transmitting
+(writing) code works by reading (without timeout) from FIFO, and if something
+is available, act accordingly. This is coupled with DIO states (e.g. to make
+sure next bit/byte on serial line is transmitted after the current).
+
+## Command multiplexing
+
+Commands, followed by arguments are filled into _Software
+Resources/CommandFIFO_. _Commands/CommandFIFOMultiplexer_ read this and
+multiplex the command to various handlers _(This is exactly how CPUs are
```
Comment (couger01) [inline README.md]:
```suggestion
multiplex the command to various handlers _(This is exactly how CPUs

```

File: README.md [hunk 3]
```diff
@@ -1,27 +1,266 @@
...
+handling instructions from binary code)_. Handlers fills in queues (e.g. ModBus
+writes queue with instructions to write if you are writing to a queue).
```
Comment (couger01) [inline README.md]:
```suggestion
creates a queue with instructions to write if you are pushing to a queue).

```

**Review by couger01 — CHANGES_REQUESTED** [review]
Couple of comments for suggested clarity otherwise LGTM.

File: README.md (hunk 3 above)
Comment (pkubanek) [reply README.md]: This is now the full paragraph:

```
Commands, followed by arguments are filled into _Software
Resources/CommandFIFO_. _Commands/CommandFIFOMultiplexer_ read this and
multiplex the command to various handlers _(This is exactly how CPUs handle
instructions from binary code)_. Handlers fills in queues (e.g. ModBus transmit
command pushes bytes to be written into a queue associated with the given bus).
**SCTL**s are handling low level IO. All this resembles a hot potato handling - 
the FPGA is pushing data through various FIFOs to make sure all incoming 
request are handled and output buffers are filled with responses.
```

**Comment by pkubanek** [comment]
Thanks for suggestions, please see if my edits matched those.

**Review by couger01 — APPROVED** [review]
LGTM.

**Pull request merged**